[Liam] Update admonition for masking Social Security Numbers (#204)
VerifiedInc/VerifiedInc.github.io: versioned_docs/version-2/reusables/prompt-user-to-confirm-their-data.mdx

You can enable or disable each input depending on whether you want the user to be able to edit the data it contains. The upside of allowing the user to edit data is that it allows them to correct any inaccuracies. The downside is that user input isn't verified and may make fraud more likely.

<Admonition type="danger" title="Mask Social Security Numbers">
    **We require you to mask Social Security Numbers because they're particularly sensitive:**

>     - Mask the first 5 digits on the server-side so that the SSN arrives at the client-side already masked.

Reviewer: ```suggestion
    - Mask the first 5 digits (server side) so your client application only handles a masked SSN.
```

>     - If you allow the user to edit the SSN, ensure validation prevents them from only modifying the last 4 digits of the masked input.

Reviewer: ```suggestion
    - If you allow the user to edit the SSN, clear the entire value when they start editing (so they don't just edit the last 4 digits).
```

Author: Liam, I updated this because clearing the entire value is a front-end-specific action. Users can copy the current value from the input and edit it directly. By stating "ensure validation prevents them from only modifying the last 4 digits of the masked input", I aimed to emphasize that this validation must occur server-side. Let me know if there’s a clearer way to convey this

Reviewer: Hmm not quite sure what you mean. The core point is that if a user edits the SSN at all, they should have to enter a full 9 digit value. That's the motivation behind clearing the whole value. So I think the frontend has to be involved here because if the SSN input just lets the user delete a couple values from the end and add new ones, that doesn't work

Author: Even if we clear the input on the front-end to make users re-enter the full SSN, someone with bad intentions could still manipulate the front-end to bypass that. That’s why I suggested adding a server-side validation step. The idea is that if the server receives a masked SSN, it could double-check the last 4 digits against the list of SSNs from the get 1-click response. This ensures the input isn’t just a partial edit of a masked value.

I know this approach is more complex, so I get that the simpler solution is to clear the input on the front-end before the user starts editing. That way, it’s clear to users that they need to re-enter the entire SSN, which solves the issue in most cases.
